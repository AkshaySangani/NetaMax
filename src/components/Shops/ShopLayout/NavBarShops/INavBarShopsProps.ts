import { IShopUser } from "dataflows/Shops/IShopUser";

export interface INavBarShopsProps {
  /**
   * The shop user data
   * @type {IShopUser}
   **/
  shopUser?: IShopUser[];
  /**
   * Handles logout action
   * @type {() => void}
   **/
  logout: () => void;
}
