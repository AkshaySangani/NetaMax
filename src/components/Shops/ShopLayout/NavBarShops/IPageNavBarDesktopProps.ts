import { IShopUser } from "dataflows/Shops/IShopUser";

export interface IPageNavBarDesktopProps {
  /**
   * The shop user data
   * @type {IShopUser}
   **/
  shopUser?: IShopUser[];
  /**
   * The WHATSAPP url
   * @type {string}
   **/
  wspUrl: string;
  /**
   * Handles logout action
   * @type {() => void}
   **/
  logout: () => void;
}
