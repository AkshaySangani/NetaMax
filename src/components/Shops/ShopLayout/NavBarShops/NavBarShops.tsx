import React, { useRef } from "react";
import { screenSizes } from "styled/screen";
import { useMediaQuery, Box, Container, Img, Flex, Button, Text } from "@chakra-ui/react";
import { INavBarShopsProps } from "./INavBarShopsProps";
import { IPageNavBarDesktopProps } from "./IPageNavBarDesktopProps";
import { useRouter } from "next/router";
import { trackEvent } from "utils/tracker";
import { SA_COPIARLIGA_CLICKED, SA_AYUDA_CLICKED, TrackerApp } from "constants/amplitudeConstants";

/**
 * The NavBar component.
 * @param {INavBarShopsProps} props The component props
 * @returns {React.ReactElement} The NavBar component
 */
export const NavBarShops = (props: INavBarShopsProps): React.ReactElement => {
  const [isDesktop] = useMediaQuery(`(min-width: ${screenSizes.mdMin}px)`);
  const sendText = "Hablar con alguien";
  const msjText = sendText.split(" ").join("%20");
  const phoneNumber = "5215545439866";
  const toWhatsApp = `https://wa.me/${phoneNumber}?text=${msjText}`;

  return (
    <Box
      sx={{ position: "sticky", top: "0" }}
      zIndex={3}
      backdropFilter="saturate(200%) blur(5px)"
      backgroundColor="rgba(255,255, 255, 1)"
      maxH={["53px", "56px"]}
    >
      <Container maxW={["container.xl", "unset"]}>
        {isDesktop ? (
          <PageNavBarDesktop {...props} wspUrl={toWhatsApp} />
        ) : (
          <PageNavBarMobile {...props} wspUrl={toWhatsApp} />
        )}
      </Container>
    </Box>
  );
};

/**
 * The NavBar shops component.
 * @param {IPageNavBarDesktopProps} props The component props
 * @returns {React.ReactElement} The NavBar shops component
 */
const PageNavBarMobile = (props: IPageNavBarDesktopProps): React.ReactElement => {
  const { wspUrl } = props;
  const router = useRouter();

  return (
    <Flex
      direction="row"
      justify="space-between"
      align="center"
      w="100%"
      p="13px 0"
      h="100%"
      maxH={["53px", "56px"]}
    >
      <Box flexDirection="column" onClick={() => router.push("/")}>
        <Img src="/assets/images/logo.png" alt="Neta logo" w={["50px", "80px"]} h="auto" />
      </Box>
      <Box flexDirection="column" onClick={() => window.open(wspUrl, "_blank")}>
        <Img src="/assets/images/misVentas/AyudaMobile.svg" alt="ayuda-logo" />
      </Box>
    </Flex>
  );
};

/**
 * The NavBar shops component.
 * @param {IPageNavBarDesktopProps} props The component props
 * @returns {React.ReactElement} The NavBar shops component
 */
const PageNavBarDesktop = (props: IPageNavBarDesktopProps): React.ReactElement => {
  const { shopUser, wspUrl } = props;
  const currentShop = shopUser && shopUser[0];
  const copyText = useRef<HTMLHeadingElement>(null);
  const shortStoreName = currentShop?.Hosts.split(".")[0];
  const router = useRouter();
  /**
   * Adds to clipboard.
   * @type {Function}
   * @returns {promise} The clipboard.
   */
  function handleCopyText() {
    trackEvent(SA_COPIARLIGA_CLICKED, [TrackerApp.Amplitude, TrackerApp.Segment], {
      storeUrl: currentShop?.Url,
    });
    if (copyText && copyText.current !== null) {
      const toCopy = copyText.current.innerText;
      // navigator clipboard api needs a secure context (https)
      if (navigator.clipboard && window.isSecureContext) {
        // navigator clipboard api method'
        return navigator.clipboard.writeText(toCopy);
      } else {
        // text area method
        const textArea = document.createElement("textarea");
        textArea.value = toCopy;
        // make the textarea out of viewport
        textArea.style.position = "fixed";
        textArea.style.left = "-999999px";
        textArea.style.top = "-999999px";
        document.body.appendChild(textArea);
        textArea.focus();
        textArea.select();
        return new Promise<void>((res, rej) => {
          // here the magic happens
          document.execCommand("copy") ? res() : rej();
          textArea.remove();
        });
      }
    } else {
      return Promise.resolve();
    }
  }

  return (
    <Flex
      direction="row"
      justify="space-between"
      align="center"
      w="100%"
      p="13px"
      h="100%"
      maxH={["53px", "56px"]}
    >
      <Box flexDirection="column" cursor="pointer" onClick={() => router.push("/")}>
        <Img src="/assets/images/logo.png" alt="Neta logo" w={["50px", "80px"]} h="auto" />
      </Box>
      <Box flexDirection="column">
        <Flex
          direction="row"
          justify="space-between"
          aling="center"
          minW="350px"
          w="auto"
          h="40px"
          p="4px 4px 4px 16px"
          bg="#F7F6F8"
          borderRadius="24px"
        >
          <Flex
            fontStyle="normal"
            fontWeight="600"
            fontSize="14px"
            align="center"
            mr="9px"
            ref={copyText}
          >
            www.neta.mx/?store={shortStoreName}
          </Flex>
          <Button
            w=" 110px"
            h="32px"
            bgColor="transparent"
            border="1px solid #353535"
            boxSizing="border-box"
            borderRadius="48px"
            fontWeight="600"
            fontSize="14px"
            color="#353535"
            cursor="pointer"
            onClick={() => handleCopyText()}
          >
            Copiar liga
          </Button>
        </Flex>
      </Box>
      <Flex direction="row" wrap="nowrap" justify="space-around" align="center">
        <Img src="/assets/images/misVentas/ShopIcon.svg" alt="Shop-icon" />
        <Text fontWeight="500" fontSize="18px" color="#5F5F5F" ml="10px">
          Tienda de {currentShop?.Name.slice(0, 20)}
        </Text>
      </Flex>
      <Box
        position="absolute"
        right="1rem"
        top="4rem"
        onClick={() => window.open(wspUrl, "_blank")}
        style={{ cursor: "pointer" }}
      >
        <Flex align="flex-start" justify="space-around">
          <Flex w="70%" direction="column" align="center">
            <Text fontWeight="600" fontSize="18px" w="100%" textAling="right">
              ¿Necesitas ayuda?
            </Text>
            <Text fontWeight="500" fontSize="18px" w="100%" textAling="right">
              Presiona el botón de ayuda
            </Text>
          </Flex>
          <Flex
            w="30%"
            align="center"
            justify="flex-end"
            onClick={() => trackEvent(SA_AYUDA_CLICKED, [TrackerApp.Amplitude, TrackerApp.Segment])}
          >
            <Img src="/assets/images/misVentas/AyudaMobile.svg" alt="ayuda-logo" />
          </Flex>
        </Flex>
      </Box>
    </Flex>
  );
};
