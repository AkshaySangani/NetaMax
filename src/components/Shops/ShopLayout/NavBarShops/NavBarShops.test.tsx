import { render, cleanup } from "@testing-library/react";
import { NavBarShops } from "./NavBarShops";
import { INavBarShopsProps } from "./INavBarShopsProps";

describe("NavBarShops component", () => {
  let mock: INavBarShopsProps;

  beforeEach(() => {
    mock = {
      shopUser: [
        {
          Id: "16031ABC22NMX",
          Name: "Store 1",
          Hosts: "Host 1",
          CompanyName: "Company 1",
          CompanyAddress: "Company Add.",
          CompanyPhoneNumber: "+140 876 313",
        },
      ],
      logout: jest.fn(),
    };
  });

  afterEach(() => {
    cleanup();
    jest.clearAllMocks();
  });

  test("NavBarShops should be render correctly", () => {
    const { container } = render(<NavBarShops {...mock} />);
    expect(container).toMatchSnapshot();
  });

  test("Name should be present in document", () => {
    mock = {
      ...mock,
      shopUser: [
        {
          Id: "16031ABC22NMX",
          Name: "Store 1",
          Hosts: "Host 1",
          CompanyName: "Company 1",
          CompanyAddress: "Company Add.",
          CompanyPhoneNumber: "+140 876 313",
        },
      ],
    };
    const { container } = render(<NavBarShops {...mock} />);
    expect(container).toMatchSnapshot();
  });
});
