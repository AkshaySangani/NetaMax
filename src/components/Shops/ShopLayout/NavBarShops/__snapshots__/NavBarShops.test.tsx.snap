// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBarShops component Name should be present in document 1`] = `
<div>
  <div
    class="css-1ezqwlw"
  >
    <div
      class="chakra-container css-1ax6rs1"
    >
      <div
        class="css-1dl3nfl"
      >
        <div
          class="css-cgq59l"
        >
          <img
            alt="Neta logo"
            class="chakra-image css-mkq9of"
            src="/assets/images/logo.png"
          />
        </div>
        <div
          class="css-cgq59l"
        >
          <img
            alt="ayuda-logo"
            class="chakra-image css-0"
            src="/assets/images/misVentas/AyudaMobile.svg"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`NavBarShops component NavBarShops should be render correctly 1`] = `
<div>
  <div
    class="css-1ezqwlw"
  >
    <div
      class="chakra-container css-1ax6rs1"
    >
      <div
        class="css-1dl3nfl"
      >
        <div
          class="css-cgq59l"
        >
          <img
            alt="Neta logo"
            class="chakra-image css-mkq9of"
            src="/assets/images/logo.png"
          />
        </div>
        <div
          class="css-cgq59l"
        >
          <img
            alt="ayuda-logo"
            class="chakra-image css-0"
            src="/assets/images/misVentas/AyudaMobile.svg"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
